tests/commands/test_patch.py

```diff
@@ -0,0 +1,63 @@
+"""
+:Description: Tests the `patch` CLI
+"""
+
+import json
+
+import pytest
+from click.testing import CliRunner
+
+from conda_recipe_manager.commands.patch import patch
+from tests.file_loading import TEST_FILES_PATH
+from tests.smoke_testing import assert_cli_usage
+
+
+def test_usage() -> None:
+    """
+    Smoke test that ensures rendering of the help menu
+    """
+    assert_cli_usage(patch)
+
+
+def test_faulty_recipe_file(fs: pytest.Function) -> None:
+    """
+    Test to check the patch operation fails with a fauly recipe i.e a recipe which does not have the correct format etc
+    """
+    runner = CliRunner()
+    fs.add_real_directory(TEST_FILES_PATH, read_only=False)  # type: ignore[attr-defined]
+
+    json_patch_path = TEST_FILES_PATH / "patch" / "correct_json_patch.json"
+    faulty_recipe_file_path = TEST_FILES_PATH / "patch" / "incorrect_recipe.yaml"
+
+    runner.invoke(patch, [str(json_patch_path), str(faulty_recipe_file_path)])
+    assert pytest.raises(IOError)
+    # assert result.exit_code == 5
+
+
+def test_faulty_json_patch_file(fs: pytest.Function) -> None:
+    """
+    Test to check the patch operation fails with a problematic json patch file
+    """
```

Comment:
Should add a `:param fs:` docstring for the sake completeness/to help others who may write tests in the future. You can copy the comment from the `http_artifact_fetcher` tests.

I won't repeat this but this applies to every function in this test.

Reply:
Why not set the `fs` type to the correct `def test_faulty_json_patch_file(fs: pyfakefs.fake_filesystem.FakeFilesystem) -> None:` Will be picked up by docs tools and editor autocomplete without needing a special docstring.

Reply:
I should have a PR up shortly that fixes this everywhere else.

Reply:
@ForgottenProgramme I'm just going to do:
```
from pyfakefs.fake_filesystem import FakeFilesystem

def test_foo(fs: FakeFilesystem) -> None:
```

Reply:
#128